packages/consumption/src/modules/requests/itemProcessors/deleteAttribute/DeleteAttributeRequestItemProcessor.ts

```diff
@@ -89,7 +90,7 @@ export class DeleteAttributeRequestItemProcessor extends GenericRequestItemProce
         requestItem: DeleteAttributeRequestItem,
         _requestInfo: LocalRequestInfo
     ): Promise<void> {
-        if (!(responseItem instanceof DeleteAttributeAcceptResponseItem)) {
+        if (responseItem instanceof AcceptResponseItem && !(responseItem instanceof DeleteAttributeAcceptResponseItem)) {
```

Comment:
Our type hints only allow for DeleteAttributeAcceptResponseItem, AcceptResponseItem and RejectResponseItem. So, it is not possible to receive a "shady ResponseItem" in this function. What I want to achieve here is that we only return early if a plain AcceptResponseItem is received (previously we had the same behavior also for RejectResponseItems). However, I can't only check `responseItem instanceof AcceptResponseItem`, since DeleteAttributeAcceptResponseItem inherits from AcceptResponseItem, so this condition will also be true if a DeleteAttributeAcceptResponseItem is received. This is why I added `&& !(responseItem instanceof DeleteAttributeAcceptResponseItem)`. Using `!(responseItem instanceof RejectResponseItem` instead of `responseItem instanceof AcceptResponseItem` as first part of the condition would yield exactly the same behavior, but the expression is easier to understand without the additional negation.

Reply:
Now I understand what you mean, then it seems to do the same thing, but it's a matter of taste what to use. So it is okay for me to leave it as it is, even though I would change it. 👍 

However, another question: why do we allow the AcceptResponseItem here at all and not only allow the use of the DeleteAttributeAcceptResponseItem and the RejectResponseItem? The other RequestItemProcessors with specific AcceptResponseItems (e.g. ReadAttributeAcceptResponseItem or ProposeAttributeAcceptResponseItem) do not allow the AcceptResponseItem.

Reply:
If an Attribute was deleted before answering the Request, a simple no-op AcceptResponseItem will be sent.

Reply:
So currently, if the Sender sends the Request for deleting an Attribute via a RelationshipTemplate, the Recipient is not known before and never ever have had the peer shared Attribute, the Sender will receive an AcceptResponseItem, even though the "true" peer still have the Attribute?